pkg/controller/acmeorders/controller.go

```diff
@@ -150,16 +183,48 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 	return c.Sync(ctx, order)
 }
 
-var keyFunc = controllerpkg.KeyFunc
+// Returns a function that finds a named Order in a particular namespace.
+func orderGetterFunc(orderLister cmacmelisters.OrderLister) func(string, string) (interface{}, error) {
+	return func(namespace, name string) (interface{}, error) {
+		return orderLister.Orders(namespace).Get(name)
+	}
+}
 
-const (
-	ControllerName = "orders"
-)
+// controllerWrapper wraps the `controller` structure to make it implement
+// the controllerpkg.queueingController interface
+type controllerWrapper struct {
+	*controller
+}
```

Comment:
I've seen this controllerWrapper pattern elsewhere in cert-manager but what's the advantage of splitting the code this way? Why not add the Register method to the `controller` ?...oh....wait....I think it's probably to make it easier to instantiate in tests, which I haven't go to yet.....

Reply:
As far as I understand (could be wrong) the wrapper just helps to make the implementation of `Register` shorter/more generic- see [without wrapper](https://github.com/jetstack/cert-manager/blob/master/pkg/controller/acmeorders/controller.go#L72) vs [with wrapper](https://github.com/jetstack/cert-manager/blob/master/pkg/controller/certificates/trigger/trigger_controller.go#L274)- by putting some of the controller creation logic into another func (`NewController`). I only added this because I saw that the newer certificate controllers all had that, so thought this is the new default and I might as well make it consistent. It is not strictly necessary for tests.

Reply:
It makes it easier to create a new controller in integration tests as it allows for splitting the functionality of `NewController` out of `Register`. There may be other ways to achieve this, but I saw that this is the pattern we use in certificates controllers. Agree that it does add complexity (as you also mention in a further comment on setting a field on embedded struct).